src/pipx/commands/upgrade.py

```diff
@@ -97,7 +97,7 @@ def upgrade_all(
         num_packages += 1
         package = venv_dir.name
         venv = Venv(venv_dir, verbose=verbose)
-        if package in skip:
+        if package in skip or "--editable" in venv.pipx_metadata.main_package.pip_args:
```

Comment:
Should probably check for the short version of `--editable`, `-e`

Reply:
I thought of that, but I think it's ok.  In `main.py` `get_pip_args()` specifically adds only `--editable` to pip args:
```
    # make sure --editable is last because it needs to be right before
    #   package specification
    if parsed_args.get("editable"):
        pip_args += ["--editable"]
```

So for both `-e` or `--editable` sent to pipx, only `--editable` ends up in `pip_args`.

Reply:
Sounds good. I guess the only other way would be if a if a user manually passes the `-e` flag to pipx's `--pip-args` argument, effectively sidestepping pipx's editable flag. I don't think that's likely though.

Reply:
Ah right about the `--pip_args`.